### PR TITLE
Add JobSink documentation

### PR DESCRIPTION
**Pull request by pierDipi**
Fixes https://github.com/knative/eventing/issues/7732

## Proposed Changes 

- Add JobSink documentation
  - Usage
  - Examples

### CODE REVIEW

File: docs/eventing/brokers/broker-developer-config-options.md
Comment (pierDipi) [inline docs/eventing/brokers/broker-developer-config-options.md:31]: Unrelated fix but I could resist and it's so wrong that time

**Comment by Cali0707** [comment]
@pierDipi do you want to wait for reviews from the users who requested the feature or should we just merge this if it is good?

File: docs/eventing/sinks/job-sink.md
Comment (pierDipi) [inline docs/eventing/sinks/job-sink.md:240]:
```suggestion
      labels:
        my-label: my-value
```

File: docs/eventing/sinks/job-sink.md
Comment (pierDipi) [inline docs/eventing/sinks/job-sink.md:277]:
```suggestion
      labels:
        my-label: my-value
```

**Comment by pierDipi** [comment]
@Cali0707 we can wait for a few days and then merge if we don't hear any feedback, we can always iterate based on the feedback we get

File: docs/eventing/sinks/job-sink.md
Comment (pierDipi) [inline docs/eventing/sinks/job-sink.md:24]:
```suggestion
When an event is sent to a `JobSink`, Eventing creates a `Job` and mounts the received event as
```

**Review by matzew — APPROVED** [review]
/lgtm
/approve

pretty neat stuff and very useful documentation!

File: docs/eventing/sinks/job-sink.md
Comment (Cali0707) [inline docs/eventing/sinks/job-sink.md:15]: Maybe mention something along the lines of `except the .metadata.name field`? As far as I know, we set the name ourselves

File: docs/eventing/sinks/job-sink.md
Comment (Cali0707) [inline docs/eventing/sinks/job-sink.md:95]: I think this section is a little confusing without some more context, because as far as I can tell we don't mention before this that the event is mounted as a file into the job

Comment (pierDipi) [reply docs/eventing/sinks/job-sink.md:95]: It's the first sentence after `# Usage` at line 24

File: docs/eventing/sinks/job-sink.md
Comment (pierDipi) [reply docs/eventing/sinks/job-sink.md:15]: I don't think this is a detail we can add here at the introduction (and also is kinda intuitive if jobs are created dynamically a name can't be fixated)

File: docs/eventing/sinks/job-sink.md
Comment (Cali0707) [reply docs/eventing/sinks/job-sink.md:95]: Ah, I guess I missed that - my bad!

**Review by Cali0707 — APPROVED** [review]
/approve

**Pull request merged**
